openai/openai.go

```diff
@@ -329,6 +331,10 @@ func fromChatRequest(r ChatCompletionRequest) (*api.ChatRequest, error) {
 			}
 			messages = append(messages, message)
 		default:
+			if msg.Role == "assistant" && msg.ToolCalls != nil {
+				messages = append(messages, api.Message{Role: msg.Role, ToolCalls: msg.ToolCalls})
```

Comment:
content and tool calls are usually mutually exclusive and the template are only written to handle one or the other. adding both to the same message is undefined

Reply:
thanks wasn't sure

Reply:
Just a note that I have seen content and tool calls both populated where Chain of Thought is used, such as by Anthropic in their responses. Not critical, just a note.

Reply:
that should be okay – this thread was in regards to ollama's internal message processing. appreciate the note